### PR TITLE
Add comment removal UI

### LINKED ISSUE #278: Moderator can remove comments
Remove is similar to delete, but slightly different. Only moderators can remove comments. Moderators can remove any comments, including their own. Moderators can also delete their own comments, hence moderators will see both "remove" and "delete" links on their own comments. On all other comments, moderators only have the remove option.

When a comment is removed, it still appears in the comment tree, with a  message saying "[comment removed]". Removed comments can be un-removed by moderators by clicking "Approve". Moderators will still be able to see the text of removed comments, although they will be grayed out.

(See mocks below)

The API issue for this is #196

image

image

**_NOTE: After the comment is removed, we will show a standard "Toast" message saying "Post Removed" (no mock for this.)_**

Removed comment as seen by moderator:
image

Removed comment as seen by user:
image

### PR DESCRIPTION
**Pull request by rhysyngsun**
#### What are the relevant tickets?
Closes #278 

#### What's this PR do?
Adds links and confirmation dialog (on remove only) for removing/adding comments.

#### How should this be manually tested?
- Be sure you don't see these options as a non-moderator user
- Verify that clicking remove and approve does what you would expect

#### What GIF best describes this PR or how it makes you feel?
remove...approve...remove...approve...

### CODE REVIEW

**Comment by aliceriot** [comment]
The functionality all seems to be good, but I see this stack trace in the console when I 'remove' a comment:

ffhfhfhasdfasdflkjfnfnfnfn

the request seems to still go through ok so I'm not sure exactly what's up

**Comment by aliceriot** [comment]
that bug is fixed!

File: static/js/containers/PostPage.js [hunk 1]
```diff
@@ -140,6 +155,48 @@ class PostPage extends React.Component<*, void> {
...
+  }
+
+  hideRemoveCommentDialog = () => {
+    const { dispatch } = this.props
+    dispatch(clearModeratingComment())
+    dispatch(hideDialog(DIALOG_REMOVE_COMMENT))
+  }
+
+  removeComment = async () => {
+    const { dispatch, removingComment } = this.props
```
Comment (aliceriot) [inline static/js/containers/PostPage.js]: `removingComment` is sort of like the focused comment, right? I might suggest rename to something like `commentToRemove` or something along those lines, the `removingComment` name to me sounds a bit more like something telling us about the status (currently removing vs no).

File: static/js/reducers/moderation.js [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+} from "../actions/moderation"
+
+import type { Action } from "../flow/reduxTypes"
+import type { CommentInTree } from "../flow/discussionTypes"
+
+export type ModerationState = {
+  comment: ?CommentInTree
+}
+
+export const INITIAL_UI_STATE: ModerationState = {
```
Comment (aliceriot) [inline static/js/reducers/moderation.js]: should this be called something like `INITIAL_MODERATION_STATE`?

Comment (rhysyngsun) [reply static/js/reducers/moderation.js]: Ooops you caught my copy/paste!

**Comment by rhysyngsun** [comment]
@aliceriot feedback addressed

**Pull request merged**